packages/11ty/_includes/components/object-card.webc

```diff
@@ -1,48 +1,61 @@
-<script webc:type="js">
-// TODO: get `ignoredAttributes` from `objects-catalog.webc`
-const ignoredAttributes = ['name', 'thumbnail', 'title'];
-const renderFields = () => {
-  return Object
-    .entries(this.webc.attributes)
-    .reduce((fields, [key, value]) => {
-      /**
-       * Only render fields for keys:
-       * - are not ignored attributes
-       * - are not objects
-       */
-      if (
-        !ignoredAttributes.includes(key) &&
-        value !== '[object Object]'
-      ) {
-        fields += `<span class="object-card__field"><span class="object-card__field-value">${value.split(',').join(', ')}</span></span>`;
-      }
-      return fields;
-    }, '');
-};
-`
-<a class="object-card" href="${url}">
-  <!--- @TODO make image and card name siblings of fields to allow them to be ordering using CSS --->
-  <figure class="object-card__image">
-    <img src="${imagePath}" alt="${objectName}">
-  </figure>
-
+<a class="object-card" href="this.webc.attributes.url">
   <div class="object-card__content">
-    <span class=object-card__title>
-      ${objectNameHtml}
-      <span class="arrow" data-outputs-exclude="epub,pdf">
-        <svg data-outputs-exclude="epub,pdf">
-          <switch>
-            <use xlink:href="#arrow-forward-icon"></use>
-          </switch>
-        </svg>
+    <template webc:for="attribute of getAttributes(this.webc.attributes, this.$data)" webc:nokeep>
+      <figure webc:if="isThumbnailAttribute(attribute)" class="object-card__image" :data-field-name="attribute.name">
+        <img :src="src(this.webc.attributes)" alt="alt(this.webc.attributes)">
+      </figure>
+      <span webc:elseif="isFirstTextAttribute(attribute, this.webc.attributes, this.$data)" class=object-card__title :data-field-name="attribute.name">
+        <span @html="attribute.value"></span>
+        <span class="arrow" data-outputs-exclude="epub,pdf">
+          <svg data-outputs-exclude="epub,pdf">
+            <switch>
+              <use xlink:href="#arrow-forward-icon"></use>
+            </switch>
+          </svg>
+        </span>
       </span>
-    </span>
-    <!--- @TODO style fields as inline-flex --->
-    <!--- ${renderFields()} --->
+      <span webc:else class="object-card__field" @html="attribute.value"></span>
+    </template>
```

Comment:
I think that this can be rendered without the conditional by preparing the data in the setup script to fit a dedicated field in the card or a `slot` and/or using a CSS first element selector. The svg icon might also be positioned using CSS flexbox or grid rather than in the `span` inline with the value.

For the display of label: value data a HTML DefinitionList may be more semantic, for example:
```
<h2>{{ objectCard.title || objectCard.name }}</h2>
<dl>
  <dt>artist</dt>
  <dd>unknown</dd>
  <dt>date</dt>
  <dd>circa 1600</dd>
  <dt>medium</dt>
  <dd>metal, wood (ebony)</dd>
</dl>
```
Which more easily allows the CSS to capitalize the first letter of the definition term `dt` element or add styles.